pkg/ocmagenthandler/ocmagenthandler_networkpolicy.go

```diff
@@ -13,22 +14,38 @@ import (
 	oah "github.com/openshift/ocm-agent-operator/pkg/consts/ocmagenthandler"
 )
 
-func buildNetworkPolicy() netv1.NetworkPolicy {
-	namespacedName := oah.BuildNamespacedName(oah.OCMAgentNetworkPolicyName)
+func buildNetworkPolicy(ocmAgent ocmagentv1alpha1.OcmAgent) netv1.NetworkPolicy {
+	var (
+		namespacedName    types.NamespacedName
+		namespaceSelector *metav1.LabelSelector
+	)
+	if ocmAgent.Spec.FleetMode {
+		namespacedName = oah.BuildNamespacedName(oah.OCMFleetAgentNetworkPolicyName)
+		namespaceSelector = &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{{
+				Key:      "name",
+				Operator: "In",
+				Values:   []string{"observatorium-*"},
```

Comment:
does `matchExpressions` support regex?

Reply:
I have tested it. It worked from my last testing. But for the second ask, I think I need to double-confirm...

Reply:
I have put the test results on the card

Reply:
Thanks for validating that. It is amazing how difficult it is to find actual documentation confirming that you can do that.